screens/onboarding/PermissionsScreen.js

```diff
@@ -0,0 +1,262 @@
+      const currStore = store;
+      currStore.distance = findDistance(this.state.region, store);
+      const [closestZip] = await geo2zip(currStore);
+      currStore.zip = closestZip;
+      sortedStores.push(currStore);
+    });
+    // sorts in place
+    sortedStores.sort(function compare(a, b) {
+      return a.distance - b.distance;
+    });
+
+    const defaultStore = stores.find((store) => {
+      return store.id === RecordIds.defaultStoreId;
+    });
+
+    const region = {
+      latitude: defaultStore.latitude,
+      longitude: defaultStore.longitude,
+      ...deltas,
+    };
+
+    // Condition for showDefaultStore requires prevState, so a little messy
+    this.setState((prevState) => ({
+      stores: sortedStores,
+      store:
+        prevState.locationErrorMsg || sortedStores[0].distance > 100
+          ? defaultStore
+          : sortedStores[0],
+      showDefaultStore: prevState.locationErrorMsg
+        ? true
+        : sortedStores[0].distance > 100,
+      region:
+        prevState.locationErrorMsg || sortedStores[0].distance > 100
+          ? region
+          : prevState.region,
+    }));
+    Analytics.setUserProperty(
+      'closest_store',
+      this.state.showDefaultStore ? 'default' : this.state.store.storeName
+    );
+  };
+
+  updateStep = () => {
+    this.setState({ step: 2 });
+  };
+
+  // The state is initially populated with stores by calling the Airtable API to get all store records
+  _populateInitialStores = async () => {
+    try {
+      const stores = await getStoreData();
+
+      // Sets list of stores in state, populates initial products
+      await this._orderStoresByDistance(stores);
+    } catch (err) {
+      console.error(
+        '[PermissionsScreen] (_populateInitialStores) Airtable:',
+        err
+      );
+      logErrorToSentry({
+        screen: 'PermissionsScreen',
+        function: '_populateInitialStores',
+        error: err,
+      });
+    }
+  };
+
+  async enableNotifications() {
+    try {
+      const customerId = await AsyncStorage.getItem('customerId');
+      await updateCustomers(customerId, {
+        notifications: true,
+      });
+
+      const r = await sendTextMessage(
+        customerId,
+        'Healthy Corners: Thank you for joining Healthy Corners notifications. Reply STOP to unsubscribe.'
+      );
+
+      console.log(r ? '[sendTextMessage] Success' : '[sendTextMessage] Failed');
+      this.navigateMapScreen();
+    } catch (err) {
+      console.error('[PermissionsScreen] (enableNotifications) Airtable:', err);
+      logErrorToSentry({
+        screen: 'PermissionsScreen',
+        action: 'enableNotifications',
+        error: err,
+      });
+    }
+  }
+
+  async navigateStoreSelect() {
+    await this._populateInitialStores();
+    this.props.navigation.navigate('StoreSelect', {
+      navigation: this.props.navigation,
+      showDefaultStore: this.state.showDefaultStore,
+      stores: this.state.stores,
```

Comment:
I get a "Non-serializable values were found in the navigation state" which isn't a huge issue, but having to pass in the list of stores makes the StoreSelect screen much harder to re-use. Also the # of params seems excessive...